addon/components/ember-wormhole.js

```diff
@@ -21,13 +24,38 @@ export default Ember.Component.extend({
         throw new Error('ember-wormhole failed to render content because the destinationElementId was set to an undefined or falsy value.');
       }
     }
-    this._morph = buffer.dom.appendMorph(destinationElement);
-    this._super.apply(this, arguments);
+    this.appendRange(destinationElement, this._firstNode, this._lastNode);
   },
 
-  willClearRender: function() {
-    var morph = this._morph;
-    Ember.run.schedule('render', morph, morph.destroy);
-    this._super.apply(this);
+  rerenderOnDestinationChange: observer('destinationElementId', function() {
+    var destinationElement = this.get('destinationElement');
+    if (destinationElement !== this._firstNode.parentNode) {
+      run.once(this, 'appendRange', destinationElement, this._firstNode, this._lastNode);
```

Comment:
don't close over state, just schedule something that will bring current state in sync with dom.